tests/e2e/test_operators.py

```diff
@@ -5,397 +5,454 @@
+            == 1
+        ), "in operator failed, string hardcoded list"
+
+    with check:
+        assert (
+            len(
+                [
+                    line
+                    for line in result.stdout.splitlines()
+                    if "test in operator #2 passed" in line
+                ]
+            )
+            == 4
+        ), "in operator failed"
+
+    with check:
+        assert (
+            len(
+                [
+                    line
+                    for line in result.stdout.splitlines()
+                    if "test contains operator #1 passed" in line
+                ]
+            )
+            == 2
+        ), "contains operator failed"
+    assert len(result.stdout.splitlines()) == 27, "Unexpected output"
+
+
+def test_logical_operators(update_environment):
     """
-    Test to validate the functionality of the negation (not) operator
+    Run a rulebook with several rules to test logical operators
     """
-
+    env = update_environment(
+        {"DEFAULT_SHUTDOWN_AFTER": str(DEFAULT_SHUTDOWN_AFTER)}
+    )
     rulebook = (
-        utils.BASE_DATA_PATH / "rulebooks/operators/test_not_operator.yml"
+        utils.BASE_DATA_PATH / "rulebooks/operators/test_logical_operators.yml"
+    )
+    vars_file = utils.BASE_DATA_PATH / "extra_vars/operator_variables.yml"
+    cmd = utils.Command(
+        rulebook=rulebook,
+        vars_file=vars_file,
+        envvars="DEFAULT_SHUTDOWN_AFTER",
     )
-    cmd = utils.Command(rulebook=rulebook)
 
     LOGGER.info(f"Running command: {cmd}")
     result = subprocess.run(
         cmd,
-        timeout=DEFAULT_TIMEOUT,
+        timeout=DEFAULT_CMD_TIMEOUT,
         capture_output=True,
         cwd=utils.BASE_DATA_PATH,
+        text=True,
+        env=env,
     )
 
-    output = utils.assert_playbook_output(result)
-
-    printed_events = [
-        line for line in output if "Event matched" in line["stdout"]
-    ]
-
-    assert len(printed_events) == 2
+    assert result.returncode == 0
+    assert not result.stderr
+
+    with check:
+        assert (
+            len(
+                [
+                    line
+                    for line in result.stdout.splitlines()
+                    if "Test and operator #1 passes" in line
+                ]
+            )
+            == 1
+        ), "logical and operator failed"
+
+    with check:
+        assert (
+            len(
+                [
+                    line
+                    for line in result.stdout.splitlines()
+                    if "Test or operator #1 passes" in line
+                ]
+            )
+            == 3
+        ), "logical or operator failed"
+
+    with check:
+        assert (
+            len(
+                [
+                    line
+                    for line in result.stdout.splitlines()
+                    if "Test and-or operator #1 passes" in line
+                ]
+            )
+            == 1
+        ), "logical and-or operator failed"
```

Comment:
Each and-or fail message should have more detail to outline which and-or test failed.